### DISCUSSION TITLE
Nginx algorithm documentation

### DISCUSSION THREAD

**Discussion by DTiunov** [nginx/nginx#311] [Show and tell]
Good day.
For my personal needs I have developed a utility to find a location that matches a given query, including sublocation levels (https://github.com/peakvisor/nginx-location-match-finder). The utility is probably not that important, but in addition I have recorded and made a graphical representation of how the nginx algorithm works (https://github.com/peakvisor/nginx-location-match-finder/tree/main/algorithms).
And here's what I was thinking: the nginx documentation is completely missing an article about how the location selection algorithm works. I could write one for you. Based on my experience, I think it would be useful.

**Comment by viktorberezikov** [comment]
It seems that the answer is in the documentation:
To find location matching a given request, nginx first checks locations defined using the prefix strings (prefix locations). Among them, the location with the longest matching prefix is selected and remembered. Then regular expressions are checked, in the order of their appearance in the configuration file. The search of regular expressions terminates on the first match, and the corresponding configuration is used. If no match with a regular expression is found then the configuration of the prefix location remembered earlier is used.
More information here:
https://nginx.org/r/location
